app/models/mixins/authentication_mixin.rb

```diff
@@ -79,7 +79,7 @@ def authentication_service_account(type = nil)
   end
 
   def required_credential_fields(_type)
-    [:userid]
+    _type.to_sym != :bearer ? [:userid] : []
```

Comment:
The `_` prefix on variable names implies that the variable is unused (and actually has some deeper meaning in Ruby syntax).  So, this should be changed to remove the `_` prefix from the type variable.